src/cdk/datepicker/datepicker-input.ts

```diff
@@ -0,0 +1,361 @@
+      // is problematic, because it would mean that it triggers another change detection cycle,
+      // which then causes a changed after checked error if the input element was focused before.
+      element.blur();
+    }
+  }
+  private _disabled: boolean;
+
+  /** Emits when a `change` event is fired on this `<input>`. */
+  @Output('cdkDatepickerChange')
+  readonly change: EventEmitter<CdkDatepickerInputEvent<D>> =
+      new EventEmitter<CdkDatepickerInputEvent<D>>();
+
+  /** Emits when an `input` event is fired on this `<input>`. */
+  @Output('cdkDatepickerInput')
+  readonly input: EventEmitter<CdkDatepickerInputEvent<D>> =
+      new EventEmitter<CdkDatepickerInputEvent<D>>();
+
+  /** Emits when the value changes (either due to user input or programmatic change). */
+  _valueChange = new EventEmitter<D | null>();
+
+  /** Emits when the disabled state has changed. */
+  _disabledChange = new EventEmitter<boolean>();
+
+  /** Implemented as part of ControlValueAccessor. */
+  _onTouched = () => {};
+
+  /** Implemented as part of ControlValueAccessor. */
+  protected _controlValAccOnChange: (value: any) => void = () => {};
+
+  /** Implemented as part of ControlValueAccessor. */
+  protected _validatorOnChange = () => {};
+
+  /** Implemented for datepicker CDK subscription. */
+  private _cdkDatepickerSubscription = Subscription.EMPTY;
+
+  /** Implemented for datepicker locale subscription. */
+  private _localeSubscription = Subscription.EMPTY;
+
+  /** The native input element to which this directive is attached. */
+  protected _inputElement: HTMLInputElement;
+
+  /** The form control validator for whether the input parses. */
+  private _parseCdkValidator: ValidatorFn = (): ValidationErrors | null => {
+    return this._lastValueValid ?
+        null : {'cdkDatepickerParse': {'text': this._elementRef.nativeElement.value}};
+  }
+
+  /** The form control validator for the min date. */
+  private _minCdkValidator: ValidatorFn = (control: AbstractControl): ValidationErrors | null => {
+    const controlValue = this._getValidDateOrNull(this._dateAdapter.deserialize(control.value));
+    return (!this.min || !controlValue ||
+        this._dateAdapter.compareDate(this.min, controlValue) <= 0) ?
+        null : {'cdkDatepickerMin': {'min': this.min, 'actual': controlValue}};
+  }
+
+  /** The form control validator for the max date. */
+  private _maxCdkValidator: ValidatorFn = (control: AbstractControl): ValidationErrors | null => {
+    const controlValue = this._getValidDateOrNull(this._dateAdapter.deserialize(control.value));
+    return (!this.max || !controlValue ||
+        this._dateAdapter.compareDate(this.max, controlValue) >= 0) ?
+        null : {'cdkDatepickerMax': {'max': this.max, 'actual': controlValue}};
+  }
+
+  /** The form control validator for the date filter. */
+  private _filterCdkValidator: ValidatorFn = (control: AbstractControl):
```

Comment:
`_filterValidator`

Reply:
`_filterValidator` is already been used in the material datepicker-input with different arguments